### DISCUSSION TITLE
"ReferenceError: describe is not defined" on next build

### DISCUSSION THREAD

**Discussion by xereda** [vercel/next.js#34249] [Help] [answered]
### Run `next info` (available from version 12.0.8 and up)

/bin/sh: 1: pnpm: not found

    Operating System:
      Platform: linux
      Arch: x64
      Version: #31-Ubuntu SMP Thu Jan 13 17:41:06 UTC 2022
    Binaries:
      Node: 14.17.3
      npm: 8.3.0
      Yarn: 1.22.10
      pnpm: N/A
    Relevant packages:
      next: 12.0.11-canary.10
      react: 17.0.2
      react-dom: 17.0.2

### What version of Next.js are you using?

12.0.11-canary.10

### What version of Node.js are you using?

v14.17.3

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu 21.10

### How are you deploying your application?

Platform not defined yet

### Describe the Bug

I'm trying to build the application, but the following error occurs:

```
 ~/d/r/nextjs-build-error    yarn build                                                                                                           qui 10 fev 2022 14:06:56
yarn run v1.22.10
$ next build
info  - Checking validity of types  
info  - Creating an optimized production build  
info  - Compiled successfully

> Build error occurred
ReferenceError: describe is not defined
    at Object.48 (/home/xereda/desenvolvimento/repositorios/nextjs-build-error/.next/server/pages/index.test.js:43:1)
    at __webpack_require__ (/home/xereda/desenvolvimento/repositorios/nextjs-build-error/.next/server/webpack-runtime.js:25:42)
    at __webpack_exec__ (/home/xereda/desenvolvimento/repositorios/nextjs-build-error/.next/server/pages/index.test.js:66:39)
    at /home/xereda/desenvolvimento/repositorios/nextjs-build-error/.next/server/pages/index.test.js:67:28
    at Object.<anonymous> (/home/xereda/desenvolvimento/repositorios/nextjs-build-error/.next/server/pages/index.test.js:70:3)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Module.require (internal/modules/cjs/loader.js:974:19) {
  type: 'ReferenceError'
}
error Command failed with exit code 1.

```

I'm using the testing library for the tests.  There are no errors in running the tests manually, as you can see in the image below:

![image](https://user-images.githubusercontent.com/14337850/153459576-54630b8e-ce02-4aea-8bed-68a0e864fb76.png)

### Expected Behavior

The expected expectation is that the build process will run successfully.

### To Reproduce

Clone the following repository, download the dependencies and run the command to build it:

https://github.com/xereda/nextjs-build-error

```
git clone git@github.com:xereda/nextjs-build-error.git
cd nextjs-build-error.git
yarn install
yarn build
```

**Comment by balazsorban44** [comment, marked as answer]
Hi, the reason is that you have a `index.test.js` file inside `pages`, and by default, all `.js` files will be included at build-time and considered as page routes. Since that file contains `describe` (which is specific to Jest tests), the build will fail. To mitigate this, you have two options:

1. move the test file outside of the `pages` directory
2. Use the `pageExtensions` config option to specify which files should be considered pages routes.
One way is to do something like this:
```js
module.exports = {
  pageExtensions: ['jsx'],
};
```

and rename all your page files to end with `.jsx` instead of `.js`

Another is to add a suffix to your pages, something like `.page`, (eg. `index.page.js`) with the following:
```js
module.exports = {
  pageExtensions: ['pages.js'],
};
```

For more information, check out our docs page! :pray: https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions#including-non-page-files-in-the-pages-directory

**Comment by stivncastillo** [comment]
I have the same error but with my components:

```
./components/Common/Button/Button.test.tsx
6:1  Error: 'describe' is not defined.  no-undef
7:3  Error: 'it' is not defined.  no-undef
9:5  Error: 'expect' is not defined.  no-undef
12:3  Error: 'it' is not defined.  no-undef
15:5  Error: 'expect' is not defined.  no-undef
18:3  Error: 'it' is not defined.  no-undef
19:21  Error: 'jest' is not defined.  no-undef
24:5  Error: 'expect' is not defined.  no-undef
```

**Comment by Sacro** [comment]
You can exclude with an exclamation mark

`pageExtensions: ['!.test.ts(x)'],`

**Reply by 0Chan-smc** [reply]
In my case excluding with exclamation mark didn't work. (Next version 13.4.3)
